bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/handler/WizLightingMediatorImpl.java

```diff
@@ -0,0 +1,254 @@
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WizLightingMediatorImpl} is responsible for receiving all the sync
+ * packets and route correctly to each handler.
+ *
+ * @author Sriram Balakrishnan - Initial contribution
+ */
+@Component(configurationPid = "WizLightingMediator", service = WizLightingMediator.class)
+@NonNullByDefault
+public class WizLightingMediatorImpl implements WizLightingMediator {
+
+    private final Logger logger = LoggerFactory.getLogger(WizLightingMediatorImpl.class);
+
+    private final Map<Thing, WizLightingHandler> handlersRegisteredByThing = new HashMap<>();
+
+    private @Nullable WizLightingUpdateReceiverRunnable receiver;
+    private @Nullable Thread wizLightingReceiverThread;
+
+    private @Nullable WizLightingDiscoveryService wizlightingDiscoveryService;
+
+    @NonNullByDefault({})
+    private NetworkAddressService networkAddressService;
+
+    /**
+     * Called at the service activation.
+     *
+     * @param componentContext the componentContext
+     */
+    protected void activate(final ComponentContext componentContext) {
+        logger.trace("Mediator has been activated by OSGI.");
+        this.initMediatorWizBulbUpdateReceiverRunnable();
+    }
+
+    /**
+     * Called at the service deactivation.
+     *
+     * @param componentContext the componentContext
+     */
+    protected void deactivate(final ComponentContext componentContext) {
+        WizLightingUpdateReceiverRunnable receiver = this.receiver;
+        if (receiver != null) {
+            receiver.shutdown();
+        }
+    }
+
+    /**
+     * This method is called by the {@link WizLightingUpdateReceiverRunnable}, when
+     * one new message has been received.
+     *
+     * @param receivedMessage the {@link WizLightingResponse} message.
+     */
+    @Override
+    public void processReceivedPacket(final WizLightingResponse receivedMessage) {
+        logger.debug("Received packet from: {} - {} with method: [{}]", receivedMessage.getWizResponseIpAddress(),
+                receivedMessage.getWizResponseMacAddress(), receivedMessage.getMethod());
+
+        String bulbIp = receivedMessage.getWizResponseIpAddress();
+        String bulbMac = receivedMessage.getWizResponseMacAddress();
+
+        if (bulbMac != "bulbMacAddress") {
+            @Nullable
+            WizLightingHandler handler = this.getHandlerRegisteredByMac(bulbMac);
+
+            if (handler != null) {
+                // deliver message to handler.
+                handler.newReceivedResponseMessage(receivedMessage);
+                logger.trace("Received message from {} delivered with success to handler of mac {}",
+                        receivedMessage.getWizResponseIpAddress(), receivedMessage.getWizResponseMacAddress());
+            } else if (bulbIp != "bulbIpAddress" && bulbMac != "bulbMacAddress") {
```

Comment:
`!=` and `==` shouldn't be used with strings. Instead use `.equals`.
```suggestion
            } else if (!"bulbIpAddress".equals(bulbIp)  && !"bulbMacAddress".equals(bulbMac) {
```

I would also like `"bulbIpAddress"` and `"bulbMacAddress"` to be static constants to be used here and in WizLightingResponse.

Reply:
In addition, you already checked if `bulbMac != "bulbMacAddress"` above so it isn't needed here.

Reply:
Fixed